### PR TITLE
Let the InvocationService drive the clock instead of having its own t…

### PR DESCRIPTION
**Pull request by baldersheim**
…icking loop.

Also use sleep_until to get intervals indendant of invoke cost as long as cost is within interval.
This also also saves a clock sample and simplifies implementation.

@havardpe PR
This replaces #21470

### CODE REVIEW

File: vespalib/src/tests/invokeservice/invokeservice_test.cpp [hunk 1]
```diff
@@ -29,6 +29,25 @@ TEST("require that wakeup is called") {
     EXPECT_EQUAL(countAtStop, a._count);
 }
 
+TEST("require that now is moving forward") {
+    InvokeCounter a;
+    InvokeServiceImpl service(1ms);
+    EXPECT_EQUAL(0u, a._count);
+    steady_time prev = steady_clock::now();
```
Comment (havardpe) [inline vespalib/src/tests/invokeservice/invokeservice_test.cpp:36]: I would prefer moving this above the creation of the invoke service to ensure now > prev also the first time

**Review by havardpe — COMMENTED** [review]
TestClock is missing...

File: staging_vespalib/src/vespa/vespalib/util/testclock.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include "clock.h"
+
+namespace vespalib {
+
+class InvokeServiceImpl;
+
+/**
```
Comment (havardpe) [inline staging_vespalib/src/vespa/vespalib/util/testclock.h:10]: this comment seems copied

Comment (baldersheim) [reply staging_vespalib/src/vespa/vespalib/util/testclock.h:10]: Fixed

File: vespalib/src/tests/invokeservice/invokeservice_test.cpp (hunk 1 above)
Comment (baldersheim) [reply vespalib/src/tests/invokeservice/invokeservice_test.cpp]: Fixed

**Pull request merged**